### PR TITLE
Remove leftover WinSock v1 checks

### PR DESCRIPTION
**Pull request by MarcelRaad**
We require Winsock 2.2 since commit 3e4b32a3e12,

### CODE REVIEW

**Review by bagder — APPROVED** [review]
Nice!

File: lib/setup-win32.h [hunk 1]
```diff
@@ -67,10 +63,6 @@
 
 #ifdef HAVE_WINSOCK2_H
 #  define USE_WINSOCK 2
-#else
-#  ifdef HAVE_WINSOCK_H
-#    error "WinSock version 1 is no longer supported, version 2 is required!"
```
Comment (mback2k) [inline lib/setup-win32.h:72]: Doesn't it make sense to keep this error condition here?

**Review by mback2k — APPROVED** [review]

File: lib/setup-win32.h (hunk 1 above)
Comment (MarcelRaad) [reply lib/setup-win32.h:72]: That would mean keeping the variable, and keeping the checks for winsock.h in the build systems. Do you think that's worth it for the case that winsock.h is available but winsock2.h isn't? I'd just treat that exactly the same as when neither is available for simplicity.

Comment (mback2k) [reply lib/setup-win32.h:72]: Fine with me, thanks for explaining it.

**Comment by MarcelRaad** [comment]
Thanks for the reviews and comments!

**Comment by mback2k** [comment]
@MarcelRaad didn't you check that all Windows CI passes at least the build stage? There are quite some compilation errors. Please take another look.

**Comment by bagder** [comment]
#7788 fixes the problem I ran into after this was merged

**Comment by MarcelRaad** [comment]
I actually did, and the change from #7788 is present in this PR, just not in what landed on master. No idea how that could happen. Sorry about that, and thanks for the fix!

**Comment by mback2k** [comment]
That fix doesn’t seem to fix the compilation errors. Mist Azure and Cirrus Windows CI builds still fail.

**Comment by bagder** [comment]
Yeah, #7788 gets lots of CI build failures on Windows with that `in_addr_t` problem. I'm thinking we need to revert that commit until we have this fixed correctly: #7790

**Pull request closed without merging**